### PR TITLE
Moved RealmBaseAdapter to io.realm.android

### PR DESCRIPTION
**Pull request by cmelchior**
Don't merge this until https://github.com/realm/realm-java/pull/1090 are ready to be merged as well.

With RealmCursor soon done #1090 and RecyclerView.Adapter #972 at some point, we will start to have quite a few classes for integration with existing Android framework classes.

In order to not clutter the global io.realm package with support classes for Android, this PR moves them to ```io.realm.android```. This will give a better separation as ```io.realm``` now only contains pure framework classes.

Note it is a breaking change as people will have to fix their imports.

@emanuelez @kneth @bmunkholm 

Note: Remember to update findbugs-filter.xml before merging

### CODE REVIEW

**Comment by kneth** [comment]
Good idea to put Android stuff in a separate package.

:+1:

File: changelog.txt [hunk 1]
```diff
@@ -1,3 +1,6 @@
+0.81
+ * BREAKING: io.realm.RealmBaseAdapter moved to io.realm.android.RealmBaseAdapter
```
Comment (emanuelez) [inline changelog.txt]: I would also add the availability of `RealmResults.getRealm()`

File: realm/src/main/java/io/realm/RealmResults.java [hunk 2]
```diff
@@ -74,7 +74,12 @@
         this.table = table;
     }
 
-    Realm getRealm() {
+    /**
+     * Return the Realm instance these query results come from.
+     *
+     * @return {@link Realm} containing the data shown.
```
Comment (emanuelez) [inline realm/src/main/java/io/realm/RealmResults.java]: This wording is a bit off

**Comment by emanuelez** [comment]
:+1: apart from the comments

**Comment by nhachicha** [comment]
:+1:

**Comment by cmelchior** [comment]
retest this please

**Comment by cmelchior** [comment]
With us moving the Android adapters to their own package this is no longer needed. Closing

**Pull request closed without merging**